test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8328,6 +8328,11 @@ def cleanup_jobs(self, extend=None, runas=None):
             select_xt = 'x'
         job_ids = self.select(extend=select_xt)
 
+        revert_server_attribs = ['managers', 'operators']
+
+        for attrib in revert_server_attribs:
+            self.manager(MGR_CMD_UNSET, SERVER, attrib, sudo=True)
```

Comment:
There is no need for the for loop.  You can pass a attrib list to manager().

Reply:
Why we are reverting server attribute in cleanup_jobs()? This method is to clean up jobs only, right.

Reply:
Since there is code relating setting server's attribute "managers" to current user, I have made changes at same place.

Reply:
@latha-subramanian had to revert the managers here so the jobs could be deleted.

Reply:
@latha-subramanian Can you tell us why you had to revert the managers here? As far as I know in Server.revert_to_defaults() cleanup jobs happen first then it reverts server attributes. So why we need to revert the managers here (I mean before reverting server attributes)?

Reply:
@hirenvadalia ,Sorry for missing this earlier. cleanup_jobs() wants mangers attribute to be set in order to delete jobs.  Some tests unset managers attribute(to test some specific scenarios) . Hence cleanup_jobs() had to set it back.  Did it answer your question?

Reply:
@latha-subramanian If I understand your answer correct then we need manager attribute set to current user to delete other user's job in PBS right? if that's the only case then I don't think we need set manager here because Server.manager() can access sudo and/or runas argument and in PBS root user can delete any user's job.

Reply:
We would also want to kill all the job processes (in _cleanup_large_num_jobs()). If I remember right, I wasn't able to do this without manager attribute set.

Reply:
@latha-subramanian Even in case of that you don't need to set manager attribute because _cleanup_large_num_jobs() has runas argument and we can pass root in that and all 3 API (Server.status, DshUtils.run_cmd and Server.delete) which are called by _cleanup_large_num_jobs can take runas argument. So I still don't think we need to set manager attribute to delete jobs before we revert server attribute.

Reply:
@hirenvadalia , Yes. I agree for the status(),run_cmd() and delete functions 'runas' argument will suffice. But there is also a qmgr call to make scheduling 'false' and 'true' in cleanup_jobs().  Is there a way out for this without setting the 'manager' attribute ?

Reply:
@hirenvadalia , I checked manager() call was successful with 'runas' as root. I will raise a bug and get this one fixed. Thanks again for pointing this out !